src/IO/H5/Helpers.hpp

```diff
@@ -113,9 +114,27 @@ bool contains_attribute(hid_t file_id, const std::string& group_name,
 
 /*!
  * \ingroup HDF5Group
- * \brief Read a DataVector from a dataset in a group
+ * \brief Open an h5 dataset and create a 'dataset_id'
```

Comment:
I don't know what you mean by `create a 'dataset_id'`. I think just having the docs say `Open and HDF5 dataset` is best since this code doesn't technically write so I'd say no creation is done.